### ISSUE TITLE
QEMU PS/2 keyboard unresponsive

### ISSUE THREAD

**Issue by selkij** [CosmosOS/Cosmos#3036]
This is just a base issue, I will investigate further later.

The keyboard doesn't seem to work with qemu, sometimes it works for half a second before not working again after a restart, so this could mean that the kernel becomes unresponsive, it is to check. Or it could be a problem with the keyboard manager directly, I haven't really checked the keyboard implementation. The keyboard works perfectly well with other media

So this is a todo list for me which will probably increase in size as I further investigate the issue.

I will also soon publish a Pull Request to fix some typos in the makefile 

- [ ] cosmos unresponsive?
- [ ] Keyboard implementation?

**Comment by 9xbt** [comment]
Cosmos moment:tm:

**Comment by ezlp** [comment]
yeah me to, but if i use ssh it seem normal. i havent try in desktop yet.

**Comment by sasdallas** [comment]
Think this is just QEMU. I have a similar issue on my OS, try launching QEMU from a different terminal and play around with it. It's finicky iirc - it is also possible & likely COSMOS isn't resetting the PS/2 controller correctly or at all

**Comment by randomusert** [comment]
> Think this is just QEMU. I have a similar issue on my OS, try launching QEMU from a different terminal and play around with it. It's finicky iirc - it is also possible & likely COSMOS isn't resetting the PS/2 controller correctly or at all

Yeah this could Be a qemu only thing since in VMware it Works just fine

**Comment by CorgiMasterEpic** [comment]
> > Think this is just QEMU. I have a similar issue on my OS, try launching QEMU from a different terminal and play around with it. It's finicky iirc - it is also possible & likely COSMOS isn't resetting the PS/2 controller correctly or at all
> 
> Yeah this could Be a qemu only thing since in VMware it Works just fine

Yeah dude i think its qemu; it works fine for me on vmware.

**Comment by sasdallas** [comment]
I no longer think it is QEMU as I have a fully working PS/2 driver.

Check: https://github.com/sasdallas/Ethereal/tree/main/drivers/misc/ps2

**Comment by randomusert** [comment]
It might Be with The qemu's version, but i don't know.
